l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Akretion - Renato Lima <renato.lima@akretion.com.br>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import requests
+
+from odoo import models, fields, api, _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+from odoo.exceptions import UserError
+
+
+class ResCurrencyRateProviderBCB(models.Model):
+    _inherit = 'res.currency.rate.provider'
+
+    service = fields.Selection(
+        selection_add=[('BCB', 'Brazilian Central Bank')])
+
+    @api.model
+    def _get_supported_currencies(self):
+        self.ensure_one()
+        if self.service == 'BCB':
```

Comment:
This may be "optimized" as in https://github.com/OCA/currency/blob/12.0/currency_rate_update/models/res_currency_rate_provider_ECB.py to avoid extra padding

Reply:
Done!

Reply:
Not sure I follow, I meant:
```
self.ensure_one()
if self.service != 'BCB':
    return super()._get_supported_currencies()

here_goes_your_code
```